osh/sh_expr_eval.py

```diff
@@ -372,11 +377,11 @@ def _StringToBigInt(self, s, blame_loc):
                 elif ch.isdigit():
                     digit = int(ch)
                 else:
-                    e_strict('Invalid digits for numeric constant %r' % digits,
-                             blame_loc)
+                    # Unreachable per the regex validation above
+                    raise AssertionError()
 
                 if digit >= base:
-                    e_strict(
+                    e_die(
```

Comment:
Changed this from `e_strict` to `e_die` because other shells already raise an error here.

Previously:

```
osh-0.23.0$ bash -c 'echo $(( 2#A ))'
bash: line 1: 2#A: value too great for base (error token is "2#A")
osh-0.23.0$ osh -c 'echo $(( 2#A ))'
0

osh-0.23.0$ osh --version
Oils 0.23.0		https://www.oilshell.org/

git commit = 30f7d6a33c655e9e32b97aabc9240ace42a11d8e
```